File: frontend/src/types/summary.ts
export interface EvidenceItem {
  id: string;
  type: string;
  label: string;
  text: string;
  numeric_values: string[];
  metadata: Record<string, unknown>;
}

export interface EvidencePackage {
  package_id: string;
  patient_id: number;
  patient_code: string;
  summary_kind: "patient" | "handoff" | string;
  evidence_items: EvidenceItem[];
  sufficient: boolean;
  insufficiency_reason: string | null;
}

export interface SummarySentence {
  index: number;
  text: string;
  evidence_ids: string[];
}

export interface SummaryResponse {
  id: string;
  patient_id: number;
  patient_code: string;
  summary_kind: "patient" | "handoff" | string;
  status: "GENERATED" | "FALLBACK" | "ABSTAINED" | "REJECTED" | string;
  validation_status: "PASSED" | "FAILED" | "ABSTAINED" | string;
  adapter_mode: string;
  model_name: string | null;
  local_only: boolean;
  text: string;
  sentences: SummarySentence[];
  evidence_package: EvidencePackage;
  validation_errors: string[];
  abstention_reason: string | null;
  created_at: string;
}

File: backend/app/schemas/summary.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class SummaryRequest(BaseModel):
    prefer_llm: bool = True


class EvidenceItemOut(BaseModel):
    id: str
    type: str
    label: str
    text: str
    numeric_values: list[str] = Field(default_factory=list)
    metadata: dict[str, Any] = Field(default_factory=dict)


class EvidencePackageOut(BaseModel):
    package_id: str
    patient_id: int
    patient_code: str
    summary_kind: str
    evidence_items: list[EvidenceItemOut]
    sufficient: bool
    insufficiency_reason: str | None = None


class SummarySentenceOut(BaseModel):
    index: int
    text: str
    evidence_ids: list[str]


class SummaryResponse(BaseModel):
    id: str
    patient_id: int
    patient_code: str
    summary_kind: str
    status: str
    validation_status: str
    adapter_mode: str
    model_name: str | None = None
    local_only: bool
    text: str
    sentences: list[SummarySentenceOut]
    evidence_package: EvidencePackageOut
    validation_errors: list[str] = Field(default_factory=list)
    abstention_reason: str | None = None
    created_at: datetime


class SummaryEvidenceResponse(BaseModel):
    summary_id: str
    evidence_package: EvidencePackageOut
